build/github/cockroach-microbench-ci.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Directories and file names
+output_dir="./artifacts/microbench"
+cleaned_current_dir="$output_dir/current"
+cleaned_base_dir="$output_dir/base"
+
+# hardcoded the file name as of now since `compare` requires a particular format for the file name.
+benchmark_file_name="pkg_sql-report.log"
+log_output_file_path="$output_dir/microbench.log"
+
+storage_bucket_url="gs://cockroach-microbench-ci"
+
+# Threshold for comparison
+threshold=$COMPARE_THRESHOLD
+exit_status=0
+
+# Configure Bazel and dev tooling
+cat <<EOF > ./.bazelrc.user
+build --config nolintonbuild
+build --remote_cache=http://127.0.0.1:9867
+test --test_tmpdir=/tmp/cockroach
+test --config crosslinux
+EOF
+
+# Exit early if SKIP_COMPARISON is true and not a push step
+if $SKIP_COMPARISON && ! $PUSH_STEP; then
+  echo "Exiting since skip comparison is enabled and this is not a push step."
+  exit $exit_status
+fi
+
+./dev doctor
+./dev build roachprod-microbench
+
+
+mkdir -p "$output_dir"
+
+# Run benchmarks and clean output
+# running count=4 here because that's the minimum required for a comparison
+./dev bench pkg/sql/tests '--filter=^BenchmarkKV$' --count 4 --verbose --bench-mem > "$log_output_file_path"
```

Comment:
I realise this whole job has a timeout, but could be good to add one here too with `--timeout=` to have some bounds on this part, so we can fail with something more specific and easily rectify.